sdk/core/Azure.Core/src/RehydrationToken.Serialization.cs

```diff
@@ -0,0 +1,121 @@
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+
+#nullable enable
+
+namespace Azure.Core
+{
+    public partial struct RehydrationToken : IJsonModel<RehydrationToken>
+    {
+        internal RehydrationToken DeserializeRehydrationToken(JsonElement element, ModelReaderWriterOptions options)
+        {
+            if (element.ValueKind == JsonValueKind.Null)
+            {
+                return default;
+            }
+            string? id = null;
+            string version = string.Empty;
+            string headerSource = string.Empty;
+            string nextRequestUri = string.Empty;
+            string initialUri = string.Empty;
+            RequestMethod requestMethod = default;
+            string? lastKnownLocation = null;
+            string finalStateVia = string.Empty;
+
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("id"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    id = property.Value.GetString()!;
+                }
+                if (property.NameEquals("version"u8))
+                {
+                    version = property.Value.GetString()!;
+                    continue;
+                }
+                if (property.NameEquals("headerSource"u8))
+                {
+                    headerSource = property.Value.GetString()!;
+                    continue;
+                }
+                if (property.NameEquals("nextRequestUri"u8))
+                {
+                    nextRequestUri = property.Value.GetString()!;
+                    continue;
+                }
+                if (property.NameEquals("initialUri"u8))
+                {
+                    initialUri = property.Value.GetString()!;
+                    continue;
+                }
+                if (property.NameEquals("requestMethod"u8))
+                {
+                    requestMethod = new RequestMethod(property.Value.GetString()!);
+                    continue;
+                }
+                if (property.NameEquals("lastKnownLocation"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    lastKnownLocation = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("finalStateVia"u8))
+                {
+                    finalStateVia = property.Value.GetString()!;
+                    continue;
+                }
+            }
+            return new RehydrationToken(id, version, headerSource, nextRequestUri, initialUri, requestMethod, lastKnownLocation, finalStateVia);
```

Comment:
What happens if one of these values is not provided in the JSON that is returned from the service?

Reply:
Same reason as the commet below.

Reply:
Added `Debug.Assert`